packages/arcgis-react/components/util/createView.tsx

```diff
@@ -39,24 +48,36 @@ export function createViewComponent<
 
     const mapContainer = useRef<HTMLDivElement>(null);
 
+    const internalId = useId();
```

Comment:
remove destructured id property and use:
```suggestion
    const internalId = useId();
    const id = mapViewProps.id ?? internalId;
```

then refactor all instances of `id ?? internalId` to `id`